source/utils/LogManager.cpp

```diff
@@ -21,3 +21,60 @@ template<> LogManager* Ogre::Singleton<LogManager>::msSingleton = nullptr;
 
 //! \brief Log filename used when OD Application throws errors without using Ogre default logger.
 const std::string LogManager::GAMELOG_NAME = "gameLog";
+
+LogManager::LogManager()
+{
+
+}
+
+LogManager::~LogManager()
+{
+
+}
+
+void LogManager::addSink(const std::shared_ptr<LogSink>& sink)
+{
+    mSinks.push_back(sink);
+}
+
+void LogManager::logMessage(LogMessageLevel level, const char* module, const char* filepath, int line, const char* format, ...)
+{
+    std::unique_lock<std::mutex>(mLock);
+
+    // timestamp
+
+    time_t current_time = ::time(0);
+    struct tm* now = ::localtime(&current_time);
+
+    char timestamp[32] = { 0 };
+    snprintf(
+        timestamp,
+        31,
+        "%02d:%02d:%02d",
+        now->tm_hour,
+        now->tm_min,
+        now->tm_sec);
+
+    // filename
+
+#if WIN32 || _WINDOWS
+    const char* path_last_slash = strrchr(filepath, '\\');
```

Comment:
Usually, we use boost for handling file path. That would avoid this ind of check that can lead to errors (for my part, I use '/' on windows as path separators and it works).